scheme/common/utils.go

```diff
@@ -35,3 +39,26 @@ func GetMandatoryPathSegment(key string, fields map[string]*structpb.Value) (str
 
 	return segment, nil
 }
+
+func GetPublicKeyFromTa(ta []byte) (crypto.PublicKey, error) {
```

Comment:
the function name is a bit misleading: since what the function does is extracting a public key from some PEM encoded data that is supposedly encoding a SubjectPublicKeyInfo.  A better name is, e.g., `DecodeSubjectPublicKeyInfo()`

Reply:
It is not just a subjectPublicKeyInfo, but a full PEM File.

I had it initially had it as:
`GetPublicKeyFromPemTa`

I am ok, to be more specific about it been a full pem ta with header as well

Reply:
> It is not just a subjectPublicKeyInfo, but a full PEM File.

hmm `x509.ParsePKIXPublicKey` only accepts SPKI, so yes it's just correctly encoded SPKIs that this function will accept.

Reply:
No, what I meant was we also check for valid PEM Headers in this function, in addition to DecodeSubjectPublicKeyInfo
hence I mentioned the name as given above!

Reply:
Or may be GetPublicKeyFromPemPublic()

Reply:
all this function is doing is providing a decoder for https://www.rfc-editor.org/rfc/rfc7468#section-13, so `DecodeTextualSPKI()` ?

Reply:
I think: GetPublicKeyFromPemPublic() precisely conveys that meaning in a simple form rather than
a cryptic use of SPKI ?

https://en.wikipedia.org/wiki/Privacy-Enhanced_Mail

Reply:
It will also be not very clear in plugin:

I think, PEM format can convey a lot of things, like Certs etc.

if we refer to:
https://www.rfc-editor.org/rfc/rfc7468#section-4

It has a clear reference to PublicKey()

So 
GetPubKeyFromPemEncodePubKey()

would be more clear

Reply:
@thomas-fossati I have renamed  it. Please have a look!

Reply:
it is ok, I think

Reply:
> it is ok, I think

we agree to disagree